hashicorp/dev-portal
# feat: use middleware for .io rewrites

## "Ready for Review" checklist




---

## Relevant links



- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](url) 🎟️

## What

Simplifies the rewrite logic by shifting it to `pages/_middleware.ts`

## Why

To support more robust e2e testing, and to eventually support the easier removal of .io rewrites.

## How

In `pages/_middleware.ts`, we examine every incoming request to determine if the request is product-specific. If so, we rewrite all URLs (minus static assets) to the respective `/_proxied-dot-io` version.

## Testing

- Go to preview URL: https://dev-portal-5xxqaf5l9-hashicorp.vercel.app/
- Change to a product
- Ensure homepage renders without issue
- Repeat for remaining products
- Switch to Nomad
- Ensure this URL returns a JSON file: https://dev-portal-5xxqaf5l9-hashicorp.vercel.app/data/vault/nomad-cluster-role.json
- 

## Anything else?




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201980045498143
